Sources/SwiftFormatRules/OrderedImports.swift

```diff
@@ -24,241 +27,365 @@ import SwiftSyntax
 ///
 /// - SeeAlso: https://google.github.io/swift#import-statements
 public final class OrderedImports: SyntaxFormatRule {
+
   public override func visit(_ node: SourceFileSyntax) -> Syntax {
-    var fileElements = [CodeBlockItemSyntax]()
-    fileElements.append(contentsOf: orderStatements(node))
-    let statements = SyntaxFactory.makeCodeBlockItemList(fileElements)
-    
-    return node.withStatements(statements)
-  }
-  
-  /// Gathers all the statements of the sourcefile, separates the imports in their appropriate
-  /// group and sorts each group by lexicographically order.
-  private func orderStatements(_ node: SourceFileSyntax) -> [CodeBlockItemSyntax] {
-    var fileComment = Trivia()
-    var impComment = Trivia()
-    var (allImports, allCode) = getAllImports(node.statements)
-
-    if allImports.count == 0 {
-      return Array(node.statements)
-    }
-    
-    if let firstImport = allImports.first,
-      let firstStatement = node.statements.first,
-      firstImport.statement == firstStatement {
-      // Extracts the comments of the imports and separates them into file comments
-      // and specific comments for the first import statement.
-      (fileComment, impComment) = getComments(node.statements.first!)
-      allImports[0].statement = replaceTrivia(
-        on: node.statements.first!,
-        token: node.statements.first?.firstToken,
-        leadingTrivia: .newlines(1) + impComment
-        ) as! CodeBlockItemSyntax
-    }
+    let lines = generateLines(codeBlockItemList: node.statements)
+
+    var imports: [Line] = []
+    var indivimports: [Line] = []
```

Comment:
For consistency with Swift naming guidelines, we might want to be a little more verbose (and camelCased) with these: `declImports` (instead of `indivimports`), `testableImports`, `codeBlocks`, `fileHeader`; and likewise for the enum cases.

Reply:
Done.